lib/puppet/provider/mongodb_replset/mongo.rb

```diff
@@ -238,8 +238,24 @@ def set_members
       Puppet.debug "New Hosts are: #{newhosts.inspect}"
 
       newhosts.each do |host|
+        retry_limit = 10
+        retry_sleep = 3
+
         output = {}
-        output = rs_arbiter == host ? rs_add_arbiter(host, master) : rs_add(host, master)
+        retry_limit.times do |n|
+          begin
```

Comment:
We can drop lines 246, 257 (new begin-end scope block is implied by the `times do |n|` loop)